module-system/module-implementations/sov-evm/src/call.rs

```diff
@@ -29,52 +28,54 @@ impl<C: sov_modules_api::Context> Evm<C> {
         _context: &C,
         working_set: &mut WorkingSet<C::Storage>,
     ) -> Result<CallResponse> {
-        let evm_tx_recovered: EvmTransactionSignedEcRecovered = tx.clone().try_into()?;
+        let evm_tx_recovered: EvmTransactionSignedEcRecovered = tx.try_into()?;
 
         let block_env = self.block_env.get(working_set).unwrap_or_default();
         let cfg = self.cfg.get(working_set).unwrap_or_default();
         let cfg_env = get_cfg_env(&block_env, cfg, None);
 
         let hash = evm_tx_recovered.hash();
-        self.transactions
-            .set(hash.as_fixed_bytes(), &tx, working_set);
 
         let evm_db: EvmDb<'_, C> = self.get_db(working_set);
 
         // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/505
         let result = executor::execute_tx(evm_db, block_env, &evm_tx_recovered, cfg_env).unwrap();
 
-        let receipt = TransactionReceipt {
+        let from = evm_tx_recovered.signer();
+        let to = evm_tx_recovered.to().map(|to| to.into());
+        let transaction = reth_rpc_types::Transaction::from_recovered(evm_tx_recovered.tx);
+
+        self.pending_transactions
+            .push(&transaction, &mut working_set.accessory_state());
+
+        let receipt = reth_rpc_types::TransactionReceipt {
             transaction_hash: hash.into(),
             // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/504
-            transaction_index: 0u64.into(),
+            transaction_index: Some(reth_primitives::U256::from(0)),
             // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/504
             block_hash: Default::default(),
             // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/504
-            block_number: Some(0u64.into()),
-            from: evm_tx_recovered.signer().into(),
-            to: evm_tx_recovered.to().map(|to| to.into()),
-            // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/504
-            cumulative_gas_used: Default::default(),
+            block_number: Some(reth_primitives::U256::from(0)),
```

Comment:
I know it is placeholder but there is a `reth_primitives::U256::ZERO`

Reply:
Discussed offline we will replace it with a real value